common/unit-test/config_unittest.cpp

```diff
@@ -0,0 +1,196 @@
+}
+
+/*
+ * Test that a broken config string will cause an exception to be thrown.
+ */
+TEST_F(StringConfigLoaderTest, UnsuccessfulLoadThrows) {
+    // This config is broken, it's missing a bracket in the group
+    const std::string configString = "[SoftwareContainer\n"
+                                     "sc.foo=bar\n";
+
+    StringConfigLoader loader(configString);
+    ASSERT_THROW(loader.loadConfig(), Glib::KeyFileError);
+}
+
+/*
+ * Test that a broken config string will cause an exception to be thrown, and that
+ * the error code thrown is the correct one.
+ */
+TEST_F(StringConfigLoaderTest, UnsuccessfulLoadThrowsExpectedErrorCode) {
+    // This config is broken, it's missing a bracket in the group
+    const std::string configString = "[SoftwareContainer\n"
+                                     "sc.foo=bar\n";
+
+    StringConfigLoader loader(configString);
+
+    try {
+        std::unique_ptr<Glib::KeyFile> config = loader.loadConfig();
+    } catch (Glib::KeyFileError &error) {
+        ASSERT_TRUE(error.code() == Glib::KeyFileError::PARSE);
+    }
+}
+
+
+/*
+ * ConfigTest suite
+ *
+ * This suite tests the Config class. It uses a stubbed loader, the StringConfigLoader, to
+ * provide the config source, and does asserts on the way Config parses the config provided.
+ */
+class ConfigTest : public ::testing::Test
+{
+};
+
+/*
+ * Test that Config contain expected string value.
+ */
+TEST_F(ConfigTest, ContainExpectedStringValue) {
+    // Well formed config
+    const std::string configString = "[SoftwareContainer]\n"
+                                     "sc.foo=bar\n"
+                                     "sc.preload=2\n";
+
+    std::unique_ptr<ConfigLoaderAbstractInterface> loader(new StringConfigLoader(configString));
+    Config config(std::move(loader));
+
+    ASSERT_TRUE(config.getStringValue("SoftwareContainer", "sc.foo") == "bar");
+}
+
+/*
+ * Test that Config throws exception on wrong config group
+ */
+TEST_F(ConfigTest, IncorrectGroupThrows) {
+    // Well formed config
+    const std::string configString = "[SoftwareContainer]\n"
+                                     "sc.foo=bar\n"
+                                     "sc.preload=2\n";
+
+    std::unique_ptr<ConfigLoaderAbstractInterface> loader(new StringConfigLoader(configString));
+    Config config(std::move(loader));
+
+    ASSERT_THROW(config.getStringValue("DoesNotExist", "sc.foo"), SCException::ConfigError);
+}
+
+/*
+ * Test that Config throws exception on wrong key
+ */
+TEST_F(ConfigTest, IncorrectKeyThrows) {
+    // Well formed config
+    const std::string configString = "[SoftwareContainer]\n"
+                                     "sc.foo=bar\n"
+                                     "sc.preload=2\n";
+
+    std::unique_ptr<ConfigLoaderAbstractInterface> loader(new StringConfigLoader(configString));
+    Config config(std::move(loader));
+
+    ASSERT_THROW(config.getStringValue("SoftwareContainer", "does-not-exist"), SCException::ConfigError);
```

Comment:
You are testing similar things here. You could consider having the test being value parameterized on group and key if there are many of these tests later on.